test/ProgressOnderwijsUtils.Tests/HtmlAttributesTest.cs

```diff
@@ -106,32 +108,74 @@ public void ResetReturnsTheSameDataAsInitiallyEvenIfConcurrentlyModified()
         public void IndexerExtractsUniquelyNamedAttr()
         {
             var div = _div._class("A")._id("B").Attribute("data-xyz", "C")._class("D");
-            IHtmlElement elem = div;
-            PAssert.That(() => elem.Attributes["id"] == "B");
+            PAssert.That(() => GetAttributes(div)["id"] == "B");
         }
 
         [Fact]
         public void IndexerExtractsFirstOccurenceWhenAmbiguousLikeTheDom()
         {
             var div = _div._class("A")._id("B").Attribute("data-xyz", "C").Attribute("data-xyz", "!!!")._class("D");
-            IHtmlElement elem = div;
-            PAssert.That(() => elem.Attributes["data-xyz"] == "C");
+            PAssert.That(() => GetAttributes(div)["data-xyz"] == "C");
         }
 
         [Fact]
         public void IndexerSupportsClassNameJoining()
         {
             var div = _div._class("A")._id("B").Attribute("data-xyz", "C").Attribute("data-xyz", "!!!")._class("D");
-            IHtmlElement elem = div;
-            PAssert.That(() => elem.Attributes["class"] == "A D");
+            PAssert.That(() => GetAttributes(div)["class"] == "A D");
+        }
+
+        [Fact]
+        public void ClassesListTwoSeparatelyAppliedAttributes()
+        {
+            var div = _div._class("A")._id("B").Attribute("data-xyz", "C").Attribute("data-xyz", "!!!")._class("D");
+            PAssert.That(() => GetAttributes(div).Classes().SequenceEqual(new[] { "A", "D" }));
+        }
+
+        [Fact]
+        public void ClassesCombinesSpaceSeparatedNamesWithSeparatelyAppliedNames()
+        {
+            var div = _div._class("A X")._id("B")._class(" D  C E  ")._class("")._class("Y");
+            PAssert.That(() => GetAttributes(div).Classes().SequenceEqual(new[] { "A", "X", "D", "C", "E", "Y" }));
+        }
+
+        [Fact]
+        public void ClassesCanBeEmpty()
+        {
+            var div = _div.Attribute("data-xyz", "C").Attribute("data-xyz", "!!!")._id("D");
+            PAssert.That(() => GetAttributes(div).Classes().None());
+        }
+
+        [Fact]
+        public void HasClassCombinesSpaceSeparatedNamesWithSeparatelyAppliedNames()
+        {
+            var div = _div._class("A X")._id("B")._class(" D  C E  ")._class("")._class("Y");
+            PAssert.That(() => GetAttributes(div).HasClass("X"));
+            PAssert.That(() => GetAttributes(div).HasClass("Y"));
+            PAssert.That(() => !GetAttributes(div).HasClass("bla"));
+            PAssert.That(() => !GetAttributes(div).HasClass("a"));
+        }
+
+        [Fact]
+        public void HasClassCornerCases()
+        {
+            var div = _div._class("A X")._id("B")._class(" D  C E  ")._class("")._class("Y");
+            PAssert.That(() => GetAttributes(div).HasClass(""), "All classlists have the empty class");
+            PAssert.That(() => !GetAttributes(div).HasClass("A X"), "You cannot check multiple classes in one call");
+        }
+
+        [Fact]
+        public void HasClassCanBeEmpty()

```

Comment:
Er wordt hieronder nergens gechecked op 'empty' toch? Dat een `_div` geen `_class` heeft opgegeven lijkt me niet de invulling van 'leeg', maar ik weet ook niet een betere term. Deze test zou ook slagen als je er wel een `_class` aan had toegevoegd die niet `D` is en dan is `_class` dus _niet_ leeg/empty.

Reply:
Naam aangepast.